tests/test_sanity.py

```diff
@@ -466,31 +465,36 @@ def test_emcc_caching(self):
       self.clear()
       output = self.do([EMCC, '-O' + str(i), '-s', '--llvm-lto', '0', path_from_root('tests', 'hello_libcxx.cpp'), '--save-bc', 'a.bc', '-s', 'DISABLE_EXCEPTION_CATCHING=0'])
       print('\n\n\n', output)
-      assert (BUILDING_MESSAGE.replace('X', libname) in output) == (i == 0), 'Must only build the first time'
+      if i == 0:
+        self.assertContained(BUILDING_MESSAGE.replace('X', libname), output)
+      else:
+        self.assertNotContained(BUILDING_MESSAGE.replace('X', libname), output)
```

Comment:
can use `assertContainedIf`

Reply:
I think I like this explicit form .... trying to think about exactly why I think that... I think its because it means there is clearly one assertion per line in this mode.   I'm don't feel strongly though.

Reply:
I think it's quicker to read on a single line. But I do see your point too.

We should probably make a decision one way or the other, and be consistent about it, though.

Reply:
Ok using assertContainedIf for now